src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -170,27 +203,53 @@ object Scalajsld {
 
       val linker = StandardImpl.linker(config)
       val logger = new ScalaConsoleLogger(options.logLevel)
-
-      val output = {
-        val js = options.output.toPath()
-        val sm = js.resolveSibling(js.getFileName().toString() + ".map")
-
-        def relURI(f: Path) =
-          new URI(null, null, f.getFileName().toString, null)
-
-        LinkerOutput(PathOutputFile(js))
-          .withSourceMap(PathOutputFile(sm))
-          .withSourceMapURI(relURI(sm))
-          .withJSFileURI(relURI(js))
-      }
-
       val cache = StandardImpl.irFileCache().newCache
 
       val result = PathIRContainer
         .fromClasspath(classpath)
         .flatMap(containers => cache.cached(containers._1))
-        .flatMap(linker.link(_, moduleInitializers, output, logger))
+        .flatMap{ irFiles =>
+          (options.output, options.outputDir) match {
+            case (Some(jsFile), None) =>
+              DeprecatedLinkerAPI().link(linker, irFiles.toList, moduleInitializers, jsFile, logger)
```

Comment:
Does this not work:

```suggestion
              (DeprecatedLinkerAPI: DeprecatedLinkerAPI).link(linker, irFiles.toList, moduleInitializers, jsFile, logger)
```

Then you won't need the additional `apply` method on `object DeprecatedLinkerAPI`.